mirrord-layer/src/pod_api.rs

```diff
@@ -22,15 +22,38 @@ struct RuntimeData {
 }
 
 impl RuntimeData {
-    async fn from_k8s(client: Client, pod_name: &str, pod_namespace: &str) -> Self {
+    async fn from_k8s(
+        client: Client,
+        pod_name: &str,
+        pod_namespace: &str,
+        container_name: &Option<String>,
+    ) -> Self {
         let pods_api: Api<Pod> = Api::namespaced(client, pod_namespace);
         let pod = pods_api.get(pod_name).await.unwrap();
         let node_name = &pod.spec.unwrap().node_name;
         let container_statuses = pod.status.unwrap().container_statuses.unwrap();
-        let container_info = container_statuses
-            .first()
-            .unwrap()
-            .container_id
+        let container_info_map = container_statuses
+            .into_iter()
+            .map(move |cs| (cs.name.clone(), cs))
+            .collect::<std::collections::HashMap<_, _>>();
+
+        let container_info = if let Some(container_name) = container_name {
+            &container_info_map
+                .get(container_name)
+                .with_context(|| {
+                    format!(
+                        "no container named {} found in namespace={}, pod={}",
+                        &container_name, &pod_namespace, &pod_name
+                    )
+                })
+                .unwrap()
+                .container_id
+        } else {
+            warn!("No container name specified, defaulting to first container found");
```

Comment:
I don't think it's a warning :)

Reply:
Ah ok. Should it just be a println! instead or just not print anything? 

Happy to change it either way!

Reply:
`info!` seems suitable.